jerry-core/api/jerry.c

```diff
@@ -5829,6 +5831,243 @@ jerry_detach_arraybuffer (const jerry_value_t value) /**< ArrayBuffer */
+  return ecma_is_shared_arraybuffer (value);
+#else /* !JERRY_BUILTIN_TYPEDARRAY */
+  JERRY_UNUSED (value);
+  return false;
+#endif /* JERRY_BUILTIN_TYPEDARRAY */
+} /* jerry_value_is_shared_arraybuffer */
+
+/**
+ * Creates a Shared ArrayBuffer object with the given length (size).
+ *
+ * Notes:
+ *      * the length is specified in bytes.
+ *      * returned value must be freed with jerry_release_value, when it is no longer needed.
+ *      * if the typed arrays are disabled this will return a TypeError.
+ *
+ * @return value of the constructed Shared ArrayBuffer object
+ */
+jerry_value_t
+jerry_create_shared_arraybuffer (const jerry_length_t size) /**< size of the Shared ArrayBuffer to create */
+{
+  jerry_assert_api_available ();
+
+#if JERRY_BUILTIN_TYPEDARRAY
+  return jerry_return (ecma_make_object_value (ecma_shared_arraybuffer_new_object (size)));
+#else /* !JERRY_BUILTIN_TYPEDARRAY */
+  JERRY_UNUSED (size);
+  return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (ecma_error_typed_array_not_supported_p)));
+#endif /* JERRY_BUILTIN_TYPEDARRAY */
+} /* jerry_create_shared_arraybuffer */
+
+/**
+ * Creates a Shared ArrayBuffer object with user specified buffer.
+ *
+ * Notes:
+ *     * the size is specified in bytes.
+ *     * the buffer passed should be at least the specified bytes big.
+ *     * if the typed arrays are disabled this will return a TypeError.
+ *     * if the size is zero or buffer_p is a null pointer this will return an empty Shared ArrayBuffer.
+ *
+ * @return value of the constructed Shared ArrayBuffer object
+ */
+jerry_value_t
+jerry_create_shared_arraybuffer_external (const jerry_length_t size, /**< size of the buffer to used */
+                                   uint8_t *buffer_p, /**< buffer to use as the Shared ArrayBuffer's backing */
+                                   jerry_value_free_callback_t free_cb) /**< buffer free callback */
+{
+  jerry_assert_api_available ();
+
+#if JERRY_BUILTIN_TYPEDARRAY
+  ecma_object_t *shared_arraybuffer;
+
+  if (JERRY_UNLIKELY (size == 0 || buffer_p == NULL))
+  {
+    shared_arraybuffer = ecma_shared_arraybuffer_new_object (0);
+  }
+  else
+  {
+    shared_arraybuffer = ecma_shared_arraybuffer_new_object_external (size, buffer_p, free_cb);
+  }
+
+  return jerry_return (ecma_make_object_value (shared_arraybuffer));
+#else /* !JERRY_BUILTIN_TYPEDARRAY */
+  JERRY_UNUSED (size);
+  JERRY_UNUSED (buffer_p);
+  JERRY_UNUSED (free_cb);
+  return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG (ecma_error_typed_array_not_supported_p)));
+#endif /* JERRY_BUILTIN_TYPEDARRAY */
+} /* jerry_create_shared_arraybuffer_external */
+
+/**
+ * Copy bytes into the Shared ArrayBuffer from a buffer.
+ *
+ * Note:
+ *     * if the object passed is not a Shared ArrayBuffer will return 0.
+ *
+ * @return number of bytes copied into the Shared ArrayBuffer.
+ */
+jerry_length_t
+jerry_shared_arraybuffer_write (const jerry_value_t value, /**< target Shared ArrayBuffer */
```

Comment:
Do we need a separate function for this or we could simply use the arraybuffer write? Same for the other helpers.

Reply:
The implementation of SharedArrayBuffer is not different from ArrayBuffer? In the future we can implement the real SharedArrayBuffer easier if we have the functions.